src/connector/src/source/pulsar/source/reader.rs

```diff
@@ -12,26 +12,95 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::collections::HashMap;
 use std::time::{SystemTime, UNIX_EPOCH};
 
 use anyhow::{anyhow, ensure, Result};
+use arrow_array::{Int32Array, Int64Array, RecordBatch};
 use async_trait::async_trait;
 use futures::StreamExt;
 use futures_async_stream::try_stream;
+use icelake::catalog::{load_catalog, CATALOG_NAME, CATALOG_TYPE};
+use icelake::io::FileScanStream;
+use icelake::{Table, TableIdentifier};
 use itertools::Itertools;
 use pulsar::consumer::InitialPosition;
 use pulsar::message::proto::MessageIdData;
 use pulsar::{Consumer, ConsumerBuilder, ConsumerOptions, Pulsar, SubType, TokioExecutor};
+use risingwave_common::array::{DataChunk, StreamChunk};
+use risingwave_common::catalog::ROWID_PREFIX;
+use risingwave_common::error::RwError;
 
+use crate::aws_utils::{ACCESS_KEY, REGION, SECRET_ACCESS};
+use crate::error::ConnectorError;
 use crate::parser::ParserConfig;
 use crate::source::pulsar::split::PulsarSplit;
 use crate::source::pulsar::{PulsarEnumeratorOffset, PulsarProperties};
 use crate::source::{
     into_chunk_stream, BoxSourceWithStateStream, Column, CommonSplitReader, SourceContextRef,
-    SourceMessage, SplitId, SplitMetaData, SplitReader,
+    SourceMessage, SplitId, SplitMetaData, SplitReader, StreamChunkWithState,
 };
 
-pub struct PulsarSplitReader {
+pub enum PulsarSplitReader {
+    Broker(PulsarBrokerReader),
+    Iceberg(PulsarIcebergReader),
+}
+
+#[async_trait]
+impl SplitReader for PulsarSplitReader {
+    type Properties = PulsarProperties;
+    type Split = PulsarSplit;
+
+    async fn new(
+        props: PulsarProperties,
+        splits: Vec<PulsarSplit>,
+        parser_config: ParserConfig,
+        source_ctx: SourceContextRef,
+        _columns: Option<Vec<Column>>,
+    ) -> Result<Self> {
+        ensure!(splits.len() == 1, "only support single split");
+        let split = splits.into_iter().next().unwrap();
+        let topic = split.topic.to_string();
+
+        tracing::debug!("creating consumer for pulsar split topic {}", topic,);
+
+        if props
+            .iceberg_loader_enabled
+            .as_ref()
+            .map(|s| s.parse::<bool>().unwrap())
```

Comment:
`parse::<bool>().unwrap()` looks problematic to me. Shouldn't panic here. (or can we use `bool` for the field? maybe + `deserialize_with = "deserialize_bool_from_string"`)